src/trainer.py

```diff
@@ -313,6 +316,10 @@ def train(self, tasks, stop_metric,
                                                         optimizer_params, scheduler_params, phase)
 
         if shared_optimizer:  # if shared_optimizer, ignore task_specific optimizers
+            if self._max_epochs_per_task:
+                n_epoch_steps = sum([info["n_tr_batches"] for info in task_infos.values()])
+                if optimizer_params["type"] == "bert_adam":
+                    optimizer_params["t_total"] = n_epoch_steps * self._max_epochs_per_task
```

Comment:
Does this behave correctly when _max_epochs_per_task is -1?

Reply:
I'm going to set max_epochs_per_task to be None by default, and also have an assert statement to check that if max_epochs_per_task is a number, it is positive

Reply:
@W4ngatang how exactly is t_total computed? Right now it's dependent on max_epoch_steps but we should make this code more generalizable so that people can use bertadams without max_epoch_steps

Reply:
we multiply the sum of `n_tr_batches` for all tasks by `max_epochs`; I don't think this behaves correctly when `max_epochs` is -1